src/stock-adjustment-mobile/edit-product-page/edit-product-page.jsx

```diff
@@ -0,0 +1,315 @@
+import { FieldArray } from 'react-final-form-arrays';
+import createDecorator from 'final-form-calculate';
+import update from 'immutability-helper';
+
+import InputField from '../../react-components/form-fields/input-field';
+import SelectField from '../../react-components/form-fields/select-field';
+import ReadOnlyField from '../../react-components/form-fields/read-only-field';
+import confirmAlertCustom from '../../react-components/modals/confirm';
+import { formatLot, formatDate, formatDateISO, toastProperties, formatProductName } from '../format-utils';
+import AddButton from '../../react-components/buttons/add-button';
+import { setAdjustment } from '../reducers/adjustment';
+import { setToastList } from '../reducers/toasts';
+
+
+const EditProductPage = ({}) => {
+    const history = useHistory();
+    const location = useLocation();
+    const dispatch = useDispatch();
+    
+    const userHomeFacility = useSelector(state => state.facilities.userHomeFacility);
+    const productOptions = useSelector(state => state.productOptions.productOptions);
+    const reasons = useSelector(state => state.reasons.reasons);
+    const adjustment = useSelector(state => state.adjustment.adjustment);
+    const program = useSelector(state => state.program.program);
+    const [productToEdit, setProductToEdit] =  useState(null)
+    const [indexOfProductToEdit, setIndexOfProductToEdit] =  useState(null);
+    const toastList = useSelector(state => state.toasts.toasts);
+    const [quantityCurrentState, setQuantityCurrentState] =  useState(null);
+    const [reasonCurrentState, setReasonCurrentState] =  useState(null);
+
+    useEffect(() => {
+        setProductToEdit(location.state.productToEdit); 
+        setIndexOfProductToEdit(location.state.indexOfProductToEdit); 
+        setQuantityCurrentState(location.state.productToEdit.quantity);
+        setReasonCurrentState(location.state.productToEdit.reason.name);
+     }, [location]);
+
+    const decorator = useMemo(() => createDecorator({
+        field: /product/,
+        updates: {
+            stockOnHand: (productVal, itemsVal) => {
+                const orderable = itemsVal.items[0]?.product ?? [];
+                if (itemsVal.items[0].hasOwnProperty('lot')) {
+                    delete itemsVal.items[0].lot;
+                } 
+                const lotCode = null; 
+                const stockOnHand = getStockOnHand(orderable, lotCode);
+                return stockOnHand;
+            }
+        }
+    },
+    {
+        field: /quantity/,
+        updates: {
+            stockOnHand: (productVal, itemsVal) => {
+                const orderable = itemsVal.items[0]?.product ?? [];
+                const lotCode = itemsVal.items[0]?.lot?.lotCode ?? null;
+                const stockOnHand = getStockOnHand(orderable, lotCode);
+                return stockOnHand;
+            }
+        }
+    }
+    ), []);
+
+    const isQuantityNotFilled = (quantity) => {
```

Comment:
Maybe this can be extracted into an util function, as I saw the same function in your previous PR.